### PR TITLE
Dialog - fullscreen size related adjustments

### PR DESCRIPTION
**Pull request by matzimowski**
# Dialog component

the product task related to this change: https://brainly.atlassian.net/browse/ANSPRIO-6337

**it's a breaking change**
- `fullscreen` prop removed
- `fullscreen` type of `size` prop added
- available sizes after this change:  `'s' | 'm' | 'l' | 'xl' | 'fullscreen'`
- **visually**: no overlay spacings (overlay is invisible)  in fullscreen size on medium and large screens

Additional changes:
- handle closing dialogs when more than 1 dialog component appear on the same page (using `Escape` key and "close" button)

## Preview

Screenshot 2022-01-31 at 17 18 07

https://user-images.githubusercontent.com/1231144/151934201-e31f979f-c4c4-48fe-9288-28fa5d9e0e66.mov

### CODE REVIEW

**Comment by coderitual** [comment]
hey @matzimowski thank you for the PR! Did you maybe verified if focus ring is working properly when 2 dialogs are opened?

I am also thinking about replacing "open another dialog" button with link - those 3 big buttons look a bit strange, wdyt?

@katarzynatobis could you please help us verify if 2 dialogs opened (one on top of another )doesn't break a11y?

**Comment by matzimowski** [comment]
@coderitual thanks for your quick response!

Yes, I did check the focus ring and IMO it works perfectly fine. But I'd really appreciate if @katarzynatobis could take a look too to confirm it 🙏 
I like your idea with the link, gonna look much better, I will replace the button with the link then 🍭

**Comment by coderitual** [comment]
hey @matzimowski, this is how focus trap behaves on my local machine. Can you confirm how it looks on your?

https://user-images.githubusercontent.com/8572321/151833684-89f3305a-2d77-4098-a49e-a0bca35f419e.mp4

File: src/components/dialog/Dialog.jsx
Comment (coderitual) [inline src/components/dialog/Dialog.jsx:116]: When we manage to fix focus trap (if we confirm it's an issue introduced by this PR), we probably wan't to remove `useEscapeKey` hook entirely.

Comment (coderitual) [reply src/components/dialog/Dialog.jsx:116]: oops I didn't notice it's already removed. Ignore

File: src/components/dialog/useFocusTrap.js
Comment (bartoszlorek) [inline src/components/dialog/useFocusTrap.js:40]: the `focusin` should fire after focusing element outside `dialogElement` so it won't trigger in this case.

**Comment by matzimowski** [comment]
> hey @matzimowski, I think focus trap is broken - this is how it behaves on my local machine. Can you confirm how it looks on yours?
> 
>  CleanShot.2022-01-31.at.17.31.37.mp4

Intersting, yea, now I can see it, it works as it's visible on your video. I'll try to investigate it.

**Comment by matzimowski** [comment]
@coderitual i believe the focus ring/trap should work fine now. i updated a new video and added it to the description.

File: src/components/dialog/useFocusTrap.js
Comment (matzimowski) [reply src/components/dialog/useFocusTrap.js]: of course 🙏 thanks @bartoszlorek. I fixed it

Comment (bartoszlorek) [reply src/components/dialog/useFocusTrap.js]: thanks, it looks ok

**Review by coderitual — APPROVED** [review]
@matzimowski thanks, works great now

**Comment by coderitual** [comment]
@matzimowski let's wait for approval from @katarzynatobis , and we're good to go

**Comment by katarzynatobis** [comment]
From accessibility point of view - everything works fine ❤️
But I've found 2 more issues:
- when you open 2nd modal, empty scrollbar is visible (that depends on your system preferences, mine is "Show scroll bars: Always")
Screenshot 2022-02-01 at 11 26 39
Screenshot 2022-02-01 at 11 26 29
- when `DialogBody` has only button(s) inside, focus ring is cropped
Screenshot 2022-02-01 at 11 25 37

**Comment by katarzynatobis** [comment]
@matzimowski I have one more. Focus trap is broken on Safari:
https://user-images.githubusercontent.com/60467496/151955439-1f1c186d-0d38-479e-80a0-5e47e29f30e2.mov
When you open 2nd dialog, focus is still trapped in the 1st one.

**Review by katarzynatobis — CHANGES_REQUESTED** [review]
@matzimowski focus trap is broken on safari:
https://user-images.githubusercontent.com/60467496/151955754-7b5f7bc7-cd32-4d02-966b-a7787fc5fb1d.mov
when you open 2nd dialog, focus is still trapped in the 1st one.

**Comment by matzimowski** [comment]
> @matzimowski focus trap is broken on safari: https://user-images.githubusercontent.com/60467496/151955754-7b5f7bc7-cd32-4d02-966b-a7787fc5fb1d.mov when you open 2nd dialog, focus is still trapped in the 1st one.

@katarzynatobis thanks a lot for testing it on your own!

Interesting, because it seems it works fine on Safari (Version 15.2 (17612.3.6.1.6)) on my end:


https://user-images.githubusercontent.com/1231144/151956414-5f121df7-f4af-4467-af57-c12a0170d1f9.mov

**Comment by katarzynatobis** [comment]
@matzimowski I think I've found the difference - focus is lost when VoiceOver is on. Looks like it's a common issue for safari & voiceover users.
@coderitual I propose to create a new issue with that bug, maybe we will be able to find a hook for that later.

**Comment by coderitual** [comment]
@katarzynatobis I agree. Probably there are even more safari issues in this regard which are not really connected to the scope of this PR and we can help fixing them later.

**Comment by matzimowski** [comment]
@katarzynatobis @coderitual would u like to me apply any more changes to this PR? 🙏

**Comment by matzimowski** [comment]
fyi @katarzynatobis @coderitual, as we discussed on the call, I adjusted `useBodyNoScroll` to handle many opened dialogs

**Review by coderitual — APPROVED** [review]

**Review by katarzynatobis — APPROVED** [review]

**Pull request merged**
